src/agents/memory/openai_session.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional
+
+from openai import AsyncOpenAI
+
+from agents.models._openai_shared import get_default_openai_client
+
+from ..items import TResponseInputItem
+from .session import SessionABC
+
+
+async def start_openai_session(openai_client: Optional[AsyncOpenAI] = None) -> str:
+    _openai_client = openai_client
+    if openai_client is None:
+        _openai_client = get_default_openai_client() or AsyncOpenAI()
+
+    response = await _openai_client.conversations.create(items=[])  # type: ignore [union-attr]
+    return response.id
+
+
+class OpenAISession(SessionABC):
+    def __init__(
+        self,
+        session_id: Optional[str] = None,
+        openai_client: Optional[AsyncOpenAI] = None,
+    ):
+        # this implementation allows to set this value later
+        self.session_id = session_id  # type: ignore
```

Comment:
using conversation_id as session_id; you can pass if you already have; otherwise, this class generates a new one under the hood